client/homescreen/activity-panel/reviews/index.js

```diff
@@ -0,0 +1,432 @@
+			<Button
+				key="delete-action"
+				isDestructive
+				isTertiary
+				onClick={ () => {
+					this.recordReviewEvent( 'delete', manageReviewEvent );
+					this.deleteReview( review.id );
+				} }
+			>
+				{ __( 'Delete', 'woocommerce-admin' ) }
+			</Button>,
+		];
+
+		return (
+			<ActivityCard
+				className="woocommerce-review-activity-card"
+				key={ review.id }
+				title={ title }
+				subtitle={ subtitle }
+				date={ review.date_created_gmt }
+				icon={ icon }
+				actions={ cardActions }
+			>
+				<span
+					dangerouslySetInnerHTML={ sanitizeHTML( review.review ) }
+				/>
+			</ActivityCard>
+		);
+	}
+
+	renderReviews( reviews ) {
+		if ( reviews.length === 0 ) {
+			return <></>;
+		}
+		return (
+			<>
+				{ reviews.map( ( review ) =>
+					this.renderReview( review, this.props )
+				) }
+				<Link
+					href={ getAdminLink(
+						'edit-comments.php?comment_type=review'
+					) }
+					onClick={ () => this.recordReviewEvent( 'reviews_manage' ) }
+					className="woocommerce-layout__activity-panel-outbound-link woocommerce-layout__activity-panel-empty"
+					type="wp-admin"
+				>
+					{ __( 'Manage all reviews', 'woocommerce-admin' ) }
+				</Link>
+			</>
+		);
+	}
+
+	render() {
+		const { isRequesting, isError, reviews } = this.props;
+
+		if ( isError ) {
+			const title = __(
+				'There was an error getting your reviews. Please try again.',
+				'woocommerce-admin'
+			);
+			const actionLabel = __( 'Reload', 'woocommerce-admin' );
+			const actionCallback = () => {
+				// @todo Add tracking for how often an error is displayed, and the reload action is clicked.
+				window.location.reload();
+			};
+
+			return (
+				<Fragment>
+					<EmptyContent
+						title={ title }
+						actionLabel={ actionLabel }
+						actionURL={ null }
+						actionCallback={ actionCallback }
+					/>
+				</Fragment>
+			);
+		}
+
+		return (
+			<Fragment>
+				<Section>
+					{ isRequesting && ! reviews.length ? (
+						<ActivityCardPlaceholder
+							className="woocommerce-review-activity-card"
+							hasAction
+							hasDate
+							lines={ 1 }
+						/>
+					) : (
+						<>{ this.renderReviews( reviews, this.props ) }</>
+					) }
+				</Section>
+			</Fragment>
+		);
+	}
+}
+
+ReviewsPanel.propTypes = {
+	reviews: PropTypes.array.isRequired,
+	isError: PropTypes.bool,
+	isRequesting: PropTypes.bool,
+};
+
+ReviewsPanel.defaultProps = {
+	reviews: [],
+	isError: false,
+	isRequesting: false,
+};
+
+ReviewsPanel.contextType = CurrencyContext;
+
+export { ReviewsPanel };
+
+export default compose( [
+	withDispatch( ( dispatch ) => {
+		const {
+			deleteReview,
+			updateReview,
+			invalidateResolutionForStoreSelector,
+		} = dispatch( REVIEWS_STORE_NAME );
+		const { createNotice } = dispatch( 'core/notices' );
+
+		const clearReviewsCache = ( emptyReviews ) => {
+			invalidateResolutionForStoreSelector( 'getReviews' );
+			if ( emptyReviews ) {
+				invalidateResolutionForStoreSelector( 'getReviewsTotalCount' );
+			}
```

Comment:
Clearing the `getReviewsTotalCount` if no reviews left, only necessary in the case we revert the last review. Without this it wouldn't update the home screen.

Reply:
This should be unnecessary after https://github.com/woocommerce/woocommerce-admin/pull/5729/commits/9d7626d558a5b0b2a1889d73bea039b4a78df6fe lands.

Reply:
@jeffstieler nice, would this require me to use `getItemsTotalCount` instead of `getReviewsTotalCount`?

Reply:
I don't think so as their implementations are identical and the total update happens in the resolver.